integration/autoupdate/e2e/main.go

```diff
@@ -0,0 +1,237 @@
+//go:build e2e
+// +build e2e
+
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/gravitational/teleport/integration/helpers/archive"
+)
+
+// Source is used for defining Teleport source directory by local ID.
+type Source struct {
+	ID   int
+	Path string
+}
+
+// VersionInfo defines what version label must be used during compilation.
+type VersionInfo struct {
+	Version  string
+	SourceID int
+}
+
+var sources = []Source{
+	// Path to Teleport source code with previous v1 version of CTMU.
+	{ID: 1, Path: "../teleport-sync"},
```

Comment:
Is this the best / easiest solution to hard code this path?

Would it be possible to add a README.md or package godoc that explains all
the test prerequisites and how to execute them, including from this note from PR description:

> The end-to-end tests use this local CDN to download the required versions for in-cluster testing. Before running the tests, the cluster must have a test user configured to simulate updates during login.